src/main/java/org/jboss/tm/usertx/TransactionListenerRegistry.java

```diff
@@ -0,0 +1,15 @@
+package org.jboss.tm.usertx;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.Transaction;
+
+public interface TransactionListenerRegistry {
+    /**
+     * @param transaction the transaction that the caller is interested in receiving events for (must not be null)
+     * @param listener an object that will be invoked when events of type {@link org.jboss.tm.usertx.TransactionEvent}
+     *                 occur
+     * @throws NotSupportedException if transaction does not support the listener
+     * feature (or the TSR resource map key is in use
+     */
+    void addListener (Transaction transaction, TransactionListener listener) throws NotSupportedException;
```

Comment:
I added a new type org.jboss.tm.listener.TransactionTypeNotSupported to 
the signature

> In src/main/java/org/jboss/tm/usertx/TransactionListenerRegistry.java 
> <https://github.com/jbosstm/jboss-transaction-spi/pull/5#discussion_r25882611>:
>
> > @@ -0,0 +1,15 @@
> > +package org.jboss.tm.usertx;
> > +
> > +import javax.transaction.NotSupportedException;
> > +import javax.transaction.Transaction;
> > +
> > +public interface TransactionListenerRegistry {
> > +    /**
> > +     * @param transaction the transaction that the caller is interested in receiving events for (must not be null)
> > +     * @param listener an object that will be invoked when events of type {@link org.jboss.tm.usertx.TransactionEvent}
> > +     *                 occur
> > +     * @throws NotSupportedException if transaction does not support the listener
> > +     * feature (or the TSR resource map key is in use
> > +     */
> > +    void addListener (Transaction transaction, TransactionListener listener) throws NotSupportedException;
>
> I think it could be useful to be able to flag an exception here in the 
> future so maybe we should just define a new Exception type?
>
> org.jboss.tm.TransactionException
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/jbosstm/jboss-transaction-spi/pull/5/files#r25882611>.
>


-- 
Michael Musgrove
Red Hat UK Ltd, Transactions Team
e: mmusgrov@redhat.com
t: +44 191 243 0870

Registered in England and Wales under Company Registration No. 03798903
Directors: Michael Cunningham (US), Charles Peters (US), Matt Parson (US), Michael O'Neill(Ireland)